### PR TITLE
Adds EnterExitTransition widget

### PR DESCRIPTION
**Pull request by abarth**
This widget smoothly resizes as its child changes and lets you provide entrance and exit transitions for its children.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie This patch is just a prototype of how we might do intro/outro animations.  Thoughts?

**Comment by Hixie** [comment]
We probably need a better story for resizing, and I think we should decide if the widget is new or not based on its key, rather than on equality of the Widget object, but otherwise seems reasonable.

File: sky/packages/sky/lib/src/rendering/overflow.dart [hunk 1]
```diff
@@ -146,6 +146,67 @@ class RenderOverflowBox extends RenderBox with RenderObjectWithChildMixin<Render
...
+  }
+
+  /// The size this render box should attempt to be.
+  Size get requestedSize => _requestedSize;
+  Size _requestedSize;
+  void set requestedSize (Size value) {
+    assert(value != null);
+    if (_requestedSize == value)
+      return;
+    _requestedSize = new Size(value.width, value.height);
```
Comment (Hixie) [inline sky/packages/sky/lib/src/rendering/overflow.dart]: move this hack to SizeObserver

**Comment by Hixie** [comment]
I think I would wrap the second argument you pass to onExit() and onEnter() in a KeyedSubtree with a GlobalObjectKey(this).

LGTM.

**Pull request merged**